### PR TITLE
Add Godot Resource Language

### PR DESCRIPTION
**Pull request by lildude**
## Description

`project.godot`, `.tres` and `.tscn` files form part of many GoDot apps and are really popular on GitHub.

I've stuck with the same colour as used by GDScript as they're related so may as well share the same colour of: `#355570`

Fixes https://github.com/github/linguist/issues/6060

## Checklist:

- [x] **I am adding a new language.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - [`.gdns`](https://github.com/search?q=path%3A*.gdns&type=code)
      - [`.gdnlib`](https://github.com/search?q=path%3A*.gdns&type=code)
      - [`.tscn`](https://github.com/search?q=path%3A*.tscn&type=code)
      - [`.tres`](https://github.com/search?q=path%3A*.tres&type=code)
      - [`project.godot`](https://github.com/search?q=path%3A*%2Fproject.godot&type=code)
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - `.tscn` and `.tres` https://github.com/Orama-Interactive/Pixelorama
      - [`project.godot`](https://github.com/ACB-prgm/Godot_GamePad/blob/9c88b449e8490f0390ec3468e67741ce9e2c912f/GamePadClient/project.godot)
      - `.gdns`, `.gdnlib` https://github.com/godot-rust/gdnative
    - Sample license(s): all covered by MIT
  - [x] I have included a syntax highlighting grammar: [URL to grammar repo]
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.
    - Not needed as they're unique

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2094,6 +2094,8 @@ GDScript:
   color: "#355570"
   extensions:
   - ".gd"
+  - ".tres"
```
Comment (Alhadis) [inline lib/linguist/languages.yml:2097]: You forgot `.uno`, `.dos` and `.cuatro`. 😉

**Review by Alhadis — APPROVED** [review]

**Review by aaronfranke — CHANGES_REQUESTED** [review]
This is completely and totally wrong.

* `.tres` and `.tscn` files are specific to Godot Engine, not to the GDScript language. Games using C# in Godot with no GDScript will still have `.tres` and `.tscn` files.

* The syntax highlighting needed for `.tres` and `.tscn` is completely different from GDScript.

**Comment by lildude** [comment]
@aaronfranke I was kinda expecting a comment like this as I know nothing about Godot and was winging it 😁

> * `.tres` and `.tscn` files are specific to Godot Engine, not to the GDScript language. Games using C# in Godot with no GDScript will still have `.tres` and `.tscn` files.

The issue here is we can't add "Godot Engine" or similar as its own language to Linguist as it isn't a language, it's framework, hence I chose GDScript which brings me to your next point:

> * The syntax highlighting needed for `.tres` and `.tscn` is completely different from GDScript.

I was motivated to add these files to GDScript as that's what the [Godot docs](https://docs.godotengine.org/en/stable/development/file_formats/tscn.html) use for the code blocks (see the page source) and things seem to look OK. 

Now I'm looking at it again, and looking at the grammar itself, would it make more sense to add a new "GDResource" data language for these extensions? That's what VSCode identifies these file as when I open them with https://github.com/godotengine/godot-vscode-plugin/ installed (which we use for the GDScript syntax highlighting).

**Comment by aaronfranke** [comment]
@lildude Yes, these files use Godot's Resource / ConfigFile format, including `project.godot`, `*.tres`, and `*.tscn`.

Note that we want to avoid marking these files as code (we should use Linguist `type: data`).

**Review by Alhadis — APPROVED** [review]
LGTM.

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -2412,6 +2412,17 @@ Go Module:
...
   ace_mode: text
   language_id: 947461016
+Godot Resource:
+  type: data
+  color: "#355570"
+  extensions:
+  - ".tres"
+  - ".tscn"
+  filenames:
+  - project.godot
```
Comment (aaronfranke) [inline lib/linguist/languages.yml:2422]: We can also add:

* Extension `".gdns"` (for GDNative)
* Extension `".gdnlib"` (for GDNative)
* Extension `".gdextension"` (for GDExtension)
* File name `override.cfg` (might be too generic of a name to put in Linguist, but we can determine if it's for Godot or not if it's next to `project.godot` in the file system, idk how to do this in Linguist).

Comment (aaronfranke) [reply lib/linguist/languages.yml:2422]: Unfortunately there is nothing that will always appear in `override.cfg`, so yeah we can just leave it out.

**Pull request merged**